jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -41,6 +41,7 @@
 import org.eclipse.jetty.util.FuturePromise;
 import org.eclipse.jetty.util.IO;
 import org.eclipse.jetty.util.Promise;
+import org.eclipse.jetty.util.Retainable;
```

Comment:
The javadoc for `Content.Souce#read` needs to be updated to acknowledge Retainable.  It needs to say that it returns a `Chunk` that has an implicit call to `retain` (or something similar).

Reply:
The javadoc mandates that the returned Chunk must be released. 
I think it's enough and does not overspecify anything about `retain()` -- it could be possible to have an implementation of `Source` that does not retain the chunks it returns.
The caller of `read()` only need to know that it has to call `release()`.

Reply:
I think saying release must be called is a good start. But i think it should say more.  it needs to say that if retain is called then release needs to be called twice.  

Even if the chunk returned is not able to be further retained, it is still a Retainable and needs to be released. So it's really just a retain limited retainable.

So i think explain the contract without referenced to retainable, but then have extra info, perhaps in an @see clause

Reply:
Done.

Reply:
Now I think you've gone too far.  I'll resolve here and make a suggestion against the latest javadoc.